sql/hive/src/test/scala/org/apache/spark/sql/hive/client/FiltersSuite.scala

```diff
@@ -108,6 +108,28 @@ class FiltersSuite extends SparkFunSuite with Logging with PlanTest {
     (a("datecol", DateType) =!= Literal(Date.valueOf("2019-01-01"))) :: Nil,
     "datecol != 2019-01-01")
 
+  filterTest("not-in int filter",
```

Comment:
Shall we add a test case for `NULL` as while, e.g. NOT IN (1, 2, ..., NULL)? Given we have bug before.

Reply:
Yes it should be, add the null value for test.

Reply:
Seems exists some issue about null value. Created [#31659](https://github.com/apache/spark/pull/31659).